### DISCUSSION TITLE
How to print SVG?

### DISCUSSION THREAD

**Discussion by pawamoy** [chalk-diagrams/chalk#136] [Q&A]
I'm trying to *print* as diagram as SVG, without writing to a file.

I found the `d.render_svg()` method, but it doesn't support printing, and doesn't understand `path="-"` as "write to standard output".

I also found the `to_svg(drawing, style)` method, and how to create these `Drawing` and `Style` instances thanks to the `chalk.backend.svg.render` function, but this just means copy/pasting `render_svg` to remove `dwg.save()` and write the drawing into a string buffer instead:

```python
pad = 0.05
height = 128
envelope = d.get_envelope()
width = int(height * envelope.width / envelope.height)
if envelope.width - width <= envelope.height - height:
    α = height / ((1 + pad) * envelope.height)
else:
    α = width / ((1 + pad) * envelope.width)
dwg = Drawing(size=(width, height))
outer = dwg.g(style="fill:white;")
marker = dwg.marker(
    id="arrow", refX=5.0, refY=1.7, size=(5, 3.5), orient="auto"
)
marker.add(dwg.polygon([(0, 0), (5, 1.75), (0, 3.5)]))
dwg.defs.add(marker)
dwg.add(outer)
s = d.center_xy().pad(1 + pad).scale(α)
e = s.get_envelope()
s = s.translate(e(-unit_x), e(-unit_y))
draw_height = max(height, width)
style = Style.root(output_size=draw_height)
outer.add(to_svg(s, dwg, style))
# for technical reasons I have to use print and not write to sys.stdout
buffer = StringIO()
dwg.write(buffer, pretty=False, indent=2)
print(buffer.getvalue())
```

Obviously I'd prefer having to call just one method or function :sweat_smile: Something like `print(d.to_svg())`.

Is there an easier way? Or would it need a feature request?

**Comment by pawamoy** [comment]
It seems I can do this instead of using a string buffer:

```python
print(dwg.tostring())
```

It's a bit better :slightly_smiling_face:

**Comment by danoneata** [comment]
Hello! We have the [`_repr_svg_`](https://github.com/chalk-diagrams/chalk/blob/master/chalk/core.py#L157) method to obtain the SVG string representation of a diagram. [This method is also used by the IPython notebook](https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html#IPython.display.display) to render the diagram. The current implementation first writes the SVG to a temporary file and then reads it; I guess that's not optimal, but it does the job?!

**Reply by pawamoy** [reply]
It does the job for sure :slightly_smiling_face: I'll use a temp file too. If you ever implement a method to just get the SVG string, let me know!
